packages/editor-ui/src/components/ValueSurvey.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+	<ModalDrawer
+		:name="VALUE_SURVEY_MODAL_KEY"
+		:beforeClose="closeDialog"
+		direction="btt"
+		width="120px"
+		class="value-survey"
+	>
+		<template slot="header">
+			<span :class="$style.title" v-text="getTitle" />
+		</template>
+		<template slot="content">
+			<section :class="$style.content">
+				<div v-if="showButtons" :class="$style.buttons">
+					<n8n-button @click="selectSurveyValue('very')" label="Very disappointed" />
+					<n8n-button @click="selectSurveyValue('somewhat')" label="Somewhat disappointed" />
+					<n8n-button @click="selectSurveyValue('not')" label="Not disappointed (it isn’t really that useful)" />
+				</div>
+				<div v-else :class="$style.email">
+					<div :class="$style.input">
+						<n8n-input
+							v-model="form.email"
+							@input="onInputChange"
+							placeholder="Your email address"
+						/>
+						<n8n-button label="Send" float="right" @click="send" :disabled="!isEmailValid" />
+					</div>
+					<div :class="$style.disclaimer">Our team would contact you personally. No spam, promise!</div>
+				</div>
+			</section>
+		</template>
+	</ModalDrawer>
+</template>
+
+<script lang="ts">
+import mixins from "vue-typed-mixins";
+import { workflowHelpers } from "@/components/mixins/workflowHelpers";
+
+import ModalDrawer from './ModalDrawer.vue';
+import { VALUE_SURVEY_MODAL_KEY } from '../constants';
+
+
+export default mixins(workflowHelpers).extend({
+	name: 'ValueSurvey',
+	components: {
+		ModalDrawer,
+	},
+	computed: {
+		getTitle (): string {
+			if (this.form.value === 'very') {
+				return this.title.very;
+			} else if (this.form.value === 'somewhat') {
+				return this.title.somewhat;
+			} else if (this.form.value === 'not') {
+				return this.title.not;
+			} else {
+				return this.title.default;
+			}
+		},
+	},
+	data() {
+		return {
+			form: {
+				email: '',
+				value: '',
+			},
+			isEmailValid: false,
+			showButtons: true,
+			title: {
+				default: "How would you feel if you could no longer use n8n?",
+				very: "Sorry to hear that. We'd love to learn how to improve. Can we reach out?",
+				somewhat: "Thanks for your feedback! We'd love to understand how we can improve. Can we reach out?",
+				not: "Great to hear! Can we reach out to see how we can make n8n even better for you?",
+			},
+			VALUE_SURVEY_MODAL_KEY,
+		};
+	},
+	methods: {
+		closeDialog(): void {
+			if (!this.isEmailValid) {
+				this.$telemetry.track('User responded value survey score', { instance_id: this.$store.getters.instanceId, how_disappointed: '' });
+			}
+			this.resetForm();
```

Comment:
not sure why this is needed. I think data will reset if we close modal and reopen it

Reply:
Removed :)